core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Analysis.scala

```diff
@@ -312,6 +312,53 @@ class Analysis(apps: Seq[ApplicationInfo]) {
     }
   }
 
+  def ioAnalysis(): Seq[IOAnalysisProfileResult] = {
+    // Filter only those apps which have dataSource Information
+    val dataSourceApps = apps.filter(_.dataSourceInfo.size > 0)
+    val allRows = dataSourceApps.flatMap { app =>
```

Comment:
why do we need to filter those with datasources?  The io metrics  include things like spill, shuffle read and write, which can happen in any stage, not just those reading  from a datasource.